### DISCUSSION TITLE
[Feature Request] A more friendly, fe and be decoupling comment admin dashboard

### DISCUSSION THREAD

**Discussion by YunYouJun** [walinejs/waline#909] [Ideas]
- 示例：[kotodama.yunyoujun.cn](https://kotodama.yunyoujun.cn) （可直接登录使用）

---

我希望能够有一个更加交互友好、前后端分离的评论管理界面。

## 原因

我看到目前的管理界面构建相对陈旧，且与后端强耦合。
在使用过程中也发现一些交互设计可能需要有所提高。
- 譬如删除评论按钮默认是隐藏的，鼠标悬浮才会出现，而在移动端我甚至需要点击一下界面方可看到按钮。
- 譬如评论每次删除需要重新刷新页面。
- ...

> https://github.com/walinejs/waline/tree/main/packages/admin

而目前强耦合的页面（譬如使用 vercel 托管），也占用了部分带宽，如若构建一个纯静态与后端分离的前端管理界面，则可以托管到 Github Pages 等地方。

## 建议

- 使用更为现代的前端构建体系进行构建，譬如 Vue3 + Vite + Element Plus + ...
- 如果 Waline 可以实现后端的解耦，并暴露提供相关的接口文档。我很乐意开展前端的构建工作。

**Comment by lizheming** [comment]
感谢反馈。当时做 Waline，前后端 UI 和界面都是我一个人在做，时间有限确实如你所说有很多不尽人意的地方。

目前后台的界面就是前后端分离的，不过目前没有接口文档，后台会用到的接口都在 https://github.com/walinejs/waline/tree/main/packages/admin/src/services 这个文件夹中，包括登录/登出，评论的获取、修改、删除，用户信息的更新等。

如果你感兴趣的话，可以直接新开个仓库做这个事情，之后我再迁移到官方仓库里来，提供 `@waline/admin` 的发布权限即可。

**Comment by YunYouJun** [comment]
我将先参照部分接口进行开发工作，大部分逻辑我可能得看代码去测试调用。
如果这部分内容可以补全文档或接口注释就再好不过啦。

**Comment by lizheming** [comment]
https://waline.js.org/api/ 已经补充了 API 文档

**Comment by YunYouJun** [comment]
Thanks.

我将会新建 repo 进行开发，并周期性在此记录进度。

### 计划主要技术栈为：

- Vue3（我看到 client 已采用此进行开发）
- Vite（更快的构建速度）
- [Nuxt3](https://nuxtjs.org/v3/)（等待预发布）
- TypeScript（参照 API 文档封装接口类型定义）
- [Element Plus](https://github.com/element-plus/element-plus)（目前为数不多的 Vue3 组件库，且过去广为应用于中后台开发，以及私心是我也是 element 团队成员 🤣 ）

### 希望达成的目标：

- 更加友好清爽的界面（toast、loading、路由状态、交互）
- 移动端自适应
- 类似管理后台的布局
- 评论与回复评论的关系展示
- 自动部署 github actions，可一键 fork 自行部署（或直接使用中心化部署后的页面）

**Comment by YunYouJun** [comment]
正在进行基础功能开发，将在完成基础功能（预览、删除、登录、退出等）实现后开源。

![image](https://user-images.githubusercontent.com/25154432/139806368-6c7c3309-a6ce-4a4a-ae8c-136e1f56f3cb.png)

![image](https://user-images.githubusercontent.com/25154432/139806414-c97f815a-9e57-4126-907e-b0c7a6f8b7c3.png)

**Comment by YunYouJun** [comment]
Waline 将评论的 Markdown 转为了富文本存储，那么编辑时则需要编辑富文本。
是否可能提供一种只是存储 Markdown 的纯文本的方式，通过实时渲染 Markdown 更为优雅？

**Comment by lizheming** [comment]
> Waline 将评论的 Markdown 转为了富文本存储，那么编辑时则需要编辑富文本。 是否可能提供一种只是存储 Markdown 的纯文本的方式，通过实时渲染 Markdown 更为优雅？

可以，之前复用了 Valine 的数据库结构，所以存的 HTML。我之前也觉得这个需要改一下… 之后处理一下

**Comment by YunYouJun** [comment]
- Demo: https://kotodama.yunyoujun.cn
- GitHub: https://github.com/YunYouJun/kotodama

已实现基础的登录、评论预览、翻页、删除功能，并开源。
目前可以简单作为删除评论使用，可能还有很多细节需要优化。

![image](https://user-images.githubusercontent.com/25154432/140611601-947e37ed-b258-4e4a-9b99-534596d7aa54.png)

> 🤣 用自己的 serverUrl 加用户名加密码应该可以直接登录使用了，有什么（除了还没做的功能的）改进建议可以告诉我～

**Comment by YunYouJun** [comment]
~~理解有误，已删除。~~

---

以及抓到个错误字 `emal` -> `email`。

![image](https://user-images.githubusercontent.com/25154432/141459605-b7a02d59-42b3-4da5-b1f0-e224de8227f0.png)

> https://waline.js.org/api/#api-User-GetToken

**Comment by YunYouJun** [comment]
- [x] 适配 `waline@2`
- [x] dark mode 适配
- [x] `ua` 字段更新为 `browser` `os` 
- [x] 回复支持 SPA

<img width="1026" alt="image" src="https://user-images.githubusercontent.com/25154432/166306289-c191e842-ac2a-49c6-8699-7798d6217a6e.png">
